crates/derive/src/combine_api.rs

```diff
@@ -177,9 +202,28 @@ impl cgo::Service for API {
                     arena,
                     out,
                     &mut state.stats.out,
+                    &mut state
+                        .enable_schema_inference
```

Comment:
could combine `enable_schema_inference` by making the inferred `Shape` an Option and then passing `self.inferred_shape.as_mut()`

Reply:
Done